Tools/PIDML/pidOnnxInterface.h

```diff
@@ -100,11 +89,9 @@ struct PidONNXInterface {
   bool applyModelBoolean(const T& track, int pid)
   {
     for (std::size_t i = 0; i < mNPids; i++) {
-      if (mModels[i * kNDetectors].mPid == pid) {
-        for (uint32_t j = 0; j < kNDetectors; j++) {
-          if (track.pt() >= mPTLimits[i][j] && (j == kNDetectors - 1 || track.pt() < mPTLimits[i][j + 1])) {
-            return mModels[i * kNDetectors + j].applyModelBoolean(track);
-          }
+      if (mModels[i].mPid == pid) {
+        if (track.pt() >= mPTLimits[i]) {
```

Comment:
So, you don't need to skip applyModelBoolean when pT is below the limit.
That will maybe improve the results.

Reply:
And, also, the mPTLimits would make now sense in PidOnnxModel, not the Interface.
Now, we will decide per-model, when what detectors it should use, instead of deciding which model to pick.